### PR TITLE
remove appendMode, intiialEntity and bidirectional plugin

### PR DESCRIPTION
**Pull request by AlvaroVega**
issue https://github.com/telefonicaid/iotagent-node-lib/issues/1413`
- [x] remove appendMode
- [x] remove createInitialEntity
- [x] remove bidirectional plugin
- [x] update tests
- [x] exclude non working tests about ngsi-ld
- [x] update doc: done in https://github.com/telefonicaid/iotagent-node-lib/pull/1437

### CODE REVIEW

**Comment by mapedraza** [comment]
This part of the doc should also be rewrited

https://github.com/telefonicaid/iotagent-node-lib/blob/4c0b7ce3b96dd33c11be1585f349146ec99e2fe5/doc/api.md?plain=1#L449-L456

**Comment by fgalan** [comment]
>  update doc

Probably this has been already accomplished (check PR https://github.com/telefonicaid/iotagent-node-lib/pull/1437)

File: lib/fiware-iotagent-lib.js [hunk 1]
```diff
@@ -350,7 +350,6 @@ exports.middlewares = middlewares;
 
 exports.dataPlugins = {
     expressionTransformation: require('./plugins/expressionPlugin'),
-    bidirectionalData: require('./plugins/bidirectionalData'),
```
Comment (fgalan) [inline lib/fiware-iotagent-lib.js:353]: File lib/plugins/bidirectionalData.js should be also removed?

Comment (AlvaroVega) [reply lib/fiware-iotagent-lib.js:353]: Done in https://github.com/telefonicaid/iotagent-node-lib/pull/1421/commits/724570d8c002f98722e793899494939b58fe8e25

File: lib/services/devices/devices-NGSI-v2.js [hunk 2]
```diff
@@ -432,7 +343,7 @@ function updateRegisterDeviceNgsi2(deviceObj, entityInfoUpdated, callback) {
                     deviceObj.subservice
                 ),
                 apply(extractDeviceDifference, deviceObj),
-                createInitialEntityNgsi2,
+                createInitialEntityNgsi2Fake,
```
Comment (fgalan) [inline lib/services/devices/devices-NGSI-v2.js:346]: Is this really needed? Why don't just remove this method from the waterfall ?

Comment (AlvaroVega) [reply lib/services/devices/devices-NGSI-v2.js:346]: * This methods makes a bypass in updateRegisterDeviceNgsi2 to allow not change
 * extractDeviceDifference and combineWithNewDevice methods

Comment (fgalan) [reply lib/services/devices/devices-NGSI-v2.js:346]: NTC

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**